osu.Game.Tests/Visual/Online/TestSceneChatLink.cs

```diff
@@ -75,9 +75,11 @@ public void Setup() => Schedule(() =>
         [TestCase("[Markdown link format with escaped [and \\[ paired] braces](https://dev.ppy.sh/home)", LinkAction.External)]
         [TestCase("(Old link format with escaped (and \\( paired) parentheses)[https://dev.ppy.sh/home] and [[also a rogue wiki link]]", LinkAction.External, LinkAction.OpenWiki)]
         [TestCase("#lobby or #osu would be blue (and work) in the ChatDisplay test (when a proper ChatOverlay is present).")] // note that there's 0 links here (they get removed if a channel is not found)
-        [TestCase("Join my multiplayer game osump://12346.", LinkAction.JoinMultiplayerMatch)]
-        [TestCase("Join my multiplayer gameosump://12346.", LinkAction.JoinMultiplayerMatch)]
-        [TestCase("Join my [multiplayer game](osump://12346).", LinkAction.JoinMultiplayerMatch)]
+        [TestCase("Join my multiplayer game osu://room/12346.", LinkAction.JoinRoom)]
+        [TestCase("Join my multiplayer gameosu://room/12346.", LinkAction.JoinRoom)]
+        [TestCase("Join my [multiplayer game](osu://room/12346).", LinkAction.JoinRoom)]
+        [TestCase("Join my multiplayer game http://dev.ppy.sh/multiplayer/rooms/12346", LinkAction.JoinRoom)]
```

Comment:
What happens if a user is trying to link a multiplayer history page for a closed room?

Reply:
If it's a "playlists room" it'll open but nothing will be interactable except for viewing results. If it's a "multiplayer room" it'll refuse to open with https://github.com/ppy/osu/blob/38af2d6ad7d241a02f4f5ce1de587c9b126dd2e7/osu.Game/OsuGame.cs#L758

Neither interaction is really new because in both playlists and multiplayer if you change filtering from "Open" to "Recently Ended" things will work in the exact same way in both.

Reply:
Hmm right, forgot we had support for that. the `room.HasEnded` check in this PR threw me off a bit.